android/app/src/main/java/co/nimblehq/smsforwarder/ui/screens/filter/AllFiltersViewModel.kt

```diff
@@ -1,86 +1,92 @@
 package co.nimblehq.smsforwarder.ui.screens.filter
 
 import androidx.hilt.lifecycle.ViewModelInject
-import co.nimblehq.smsforwarder.domain.data.Filter
 import co.nimblehq.smsforwarder.domain.data.IncomingSmsEntity
 import co.nimblehq.smsforwarder.domain.usecase.ForwardIncomingSmsUseCase
+import co.nimblehq.smsforwarder.domain.usecase.GetFiltersUseCase
 import co.nimblehq.smsforwarder.domain.usecase.ObserveIncomingSmsUseCase
+import co.nimblehq.smsforwarder.domain.usecase.ObserveFiltersUseCase
 import co.nimblehq.smsforwarder.ui.base.BaseViewModel
 import co.nimblehq.smsforwarder.ui.base.NavigationEvent
 import io.reactivex.Observable
 import io.reactivex.rxkotlin.subscribeBy
 import io.reactivex.subjects.BehaviorSubject
 
-interface AllFiltersViewModel {
-    fun navigateToFilterManager()
-}
-
 interface Input {
     fun refresh()
 }
 
-class AllFiltersViewModelImpl @ViewModelInject constructor(
+class AllFiltersViewModel @ViewModelInject constructor(
     private val observeIncomingSmsUseCase: ObserveIncomingSmsUseCase,
-    private val forwardIncomingSmsUseCase: ForwardIncomingSmsUseCase
-) : BaseViewModel(), AllFiltersViewModel, Input {
+    private val forwardIncomingSmsUseCase: ForwardIncomingSmsUseCase,
+    private val observeFiltersUseCase: ObserveFiltersUseCase,
+    private val getFiltersUseCase: GetFiltersUseCase
+) : BaseViewModel(), Input {
 
     val input: Input = this
 
-    private val _data = BehaviorSubject.create<List<Filter>>()
-    val data: Observable<List<Filter>>
+    private val _data = BehaviorSubject.create<List<FilterUiModel>>()
+    val data: Observable<List<FilterUiModel>>
         get() = _data
 
     init {
+        observeFilters()
+        getFilters()
         observeIncomingSms()
-        _data.onNext(
-            listOf(
-                Filter(
-                    sender = "Lydia",
-                    forwardEmailAddress = "lydia@test.com",
-                    forwardSlackChannel = "sms-forwarder",
-                    template = ""
-                ),
-                Filter(
-                    sender = "J.",
-                    forwardEmailAddress = "",
-                    forwardSlackChannel = "sms-forwarder",
-                    template = ""
-                )
-            )
-        )
     }
 
     // TODO Remove this
     override fun refresh() {
         observeIncomingSms()
     }
 
-    override fun navigateToFilterManager() {
+    fun navigateToFilterManager() {
```

Comment:
Actually, the fragment should not access the ViewModel method directly. 🤔

Reply:
@lydiasama You mean we should add the `Output`? 🤔

Reply:
It is not an output. That's why as an old implementation we use the `AllFiltersViewModel` interface. 🤔  I'm not sure about this convention in android-template. What do you think?